docs/scripts/api/buildInterfacesDocumentation.ts

```diff
@@ -51,224 +73,153 @@ export function generateProperties(
-    const comment = signature?.comment || propertyReflection.comment;
-    const description = linkify(comment?.shortText, documentedInterfaces, 'markdown');
+  object.properties.forEach((property) => {
+    const defaultValue = property.tags.default?.text?.[0].text;
 
-    if (propertyReflection.flags.isOptional) {
-      name = `<span class="prop-name optional">${name}<sup><abbr title="optional">?</abbr></sup></span>`;
-    } else {
-      name = `<span class="prop-name">${name}</span>`;
-    }
+    const formattedName = property.isOptional
+      ? `<span class="prop-name optional">${property.name}<sup><abbr title="optional">?</abbr></sup></span>`
+      : `<span class="prop-name">${property.name}</span>`;
 
-    let defaultValue = '';
-    const defaultTag = comment && comment.getTag('default');
-    if (defaultTag) {
-      defaultValue = `<span class="prop-default">${escapeCell(defaultTag.text)}</span>`;
-    }
+    const formattedType = `<span class="prop-type">${escapeCell(property.typeStr)}</span>`;
 
-    let typeFormatted = '<span class="prop-type">';
-    if (signature) {
-      typeFormatted += escapeCell(generateSignatureStr(signature));
-    } else if (type) {
-      typeFormatted += escapeCell(generateTypeStr(type));
-    }
-    typeFormatted += '</span>';
+    const formattedDefaultValue =
+      defaultValue == null ? '' : `<span class="prop-default">${escapeCell(defaultValue)}</span>`;
+
+    const formattedDescription = escapeCell(
+      linkify(property.description, documentedInterfaces, 'markdown'),
+    );
 
     if (hasDefaultValue) {
-      text += `| ${name} | ${typeFormatted} | ${defaultValue} | ${escapeCell(description)} |\n`;
+      text += `| ${formattedName} | ${formattedType} | ${formattedDefaultValue} | ${formattedDescription} |\n`;
     } else {
-      text += `| ${name} | ${typeFormatted} | ${escapeCell(description)} |\n`;
+      text += `| ${formattedName} | ${formattedType} | ${formattedDescription} |\n`;
     }
   });
 
   return text;
 }
 
-function generateImportStatement(declaration: DeclarationContext) {
+function generateImportStatement(object: ParsedObject) {
   // TODO: Check if interface was exported
-  if (declaration.name === 'GridApi') {
+  if (object.name === 'GridApi') {
     return `\`\`\`js
-import { ${declaration.name} } from '@mui/x-data-grid-pro';
+import { ${object.name} } from '@mui/x-data-grid-pro';
 \`\`\``;
   }
 
   return `\`\`\`js
-import { ${declaration.name} } from '@mui/x-data-grid-pro';
+import { ${object.name} } from '@mui/x-data-grid-pro';
 // or
-import { ${declaration.name} } from '@mui/x-data-grid';
+import { ${object.name} } from '@mui/x-data-grid';
 \`\`\``;
 }
 
-function generateMarkdown(
-  declaration: DeclarationContext,
-  documentedInterfaces: Map<string, boolean>,
-) {
+function generateMarkdown(object: ParsedObject, documentedInterfaces: Map<string, boolean>) {
   return [
-    `# ${declaration.name} Interface`,
+    `# ${object.name} Interface`,
```

Comment:
The next step would be to use the name of the interface with generics, e.g. `GridCellParams<V = any, R = any, F = V>`.

Reply:
Yes I have a lot of ideas to improve the output